### ISSUE TITLE
tagme() fails for tag projects without tags

### ISSUE THREAD

**Issue by j-sayers** [MotusDev/Motus-TO-DO#1189]
On motus version 5.0 calling `tagme()` on a tag project fails if the project doesn't have tags. @steffilazerte 

> library(motus)
> packageVersion('motus')
[1] ‘5.0.0’
> d <- 516 # project that doesn't yet have tags
> tagme(d, dir = 'C:/MotusData')
Checking for new data in project 516
Updating metadata
activity:     0 new batch records to check
nodeData:     0 new batch records to check
Fetching deprecated batches
Error in `dplyr::anti_join()`:
! Join columns must be present in data.
x Problem with `batchID`.
Run `rlang::last_error()` to see where the error occurred.

**Comment by j-sayers** [comment]
This appears to not just affect tag projects without registered tags, but also some receivers. EG CTT-8E20667AE036 

> 
> Checking for new data for receiver CTT-8E20667AE036 (deviceID: 2720)
> Receiver CTT-8E20667AE036:  got    12 batch records
> batchID  5678388 (#     1 of     12): got     23 runs starting at               0
> batchID  5678388 (#     1 of     12): got    388 hits starting at               0
> batchID  5678388 (#     1 of     12): got    287 pulse counts                     
> batchID  5678389 (#     2 of     12): got     76 runs starting at               0
> batchID  5678389 (#     2 of     12): got   1603 hits starting at               0
> batchID  5678389 (#     2 of     12): got   1704 pulse counts                     
> batchID  5678390 (#     3 of     12): got   1678 pulse counts                     
> batchID  5678391 (#     4 of     12): got   1776 pulse counts                     
> batchID  5678392 (#     5 of     12): got   2048 pulse counts                     
> batchID  5678393 (#     6 of     12): got    372 pulse counts                     
> batchID  5678394 (#     7 of     12): got   1401 pulse counts                     
> batchID  5678395 (#     8 of     12): got    628 pulse counts                     
> batchID  5678396 (#     9 of     12): got   1200 pulse counts                     
> batchID  5678397 (#    10 of     12): got    153 pulse counts                     
> batchID  5678398 (#    11 of     12): got    366 pulse counts                     
> batchID  5678399 (#    12 of     12): got      1 runs starting at               0
> batchID  5678399 (#    12 of     12): got      2 hits starting at               0
> batchID  5678399 (#    12 of     12): got    514 pulse counts                     
> activity:    12 new batch records to check
> batchID  5678388 (#     1 of     12): got      5 activity records
> batchID  5678388 (#     1 of     12): got      0 activity records
> batchID  5678389 (#     2 of     12): got     33 activity records
> batchID  5678389 (#     2 of     12): got      0 activity records
> batchID  5678390 (#     3 of     12): got      0 activity records
> batchID  5678391 (#     4 of     12): got      0 activity records
> batchID  5678392 (#     5 of     12): got      0 activity records
> batchID  5678393 (#     6 of     12): got      0 activity records
> batchID  5678394 (#     7 of     12): got      0 activity records
> batchID  5678395 (#     8 of     12): got      0 activity records
> batchID  5678396 (#     9 of     12): got      0 activity records
> batchID  5678397 (#    10 of     12): got      0 activity records
> batchID  5678398 (#    11 of     12): got      0 activity records
> batchID  5678399 (#    12 of     12): got      1 activity records
> batchID  5678399 (#    12 of     12): got      0 activity records
> Downloaded 39 activity records
> nodeData:     0 new batch records to check
> Fetching deprecated batches
> Error in `dplyr::anti_join()`:
> ! Join columns must be present in data.
> x Problem with `batchID`.
> Backtrace:
>  1. motus::tagme(...)
>  6. dplyr:::anti_join.data.frame(b, ., by = "batchID")

**Comment by steffilazerte** [comment]
Yes, the problem isn't related to the number of hits or tags, but the number of deprecated batches.  I have a fix prepped on the `dev` branch but am waiting for Catherine to fix an API issue before I merge it in. 

For now you should be able to use `tagme(..., skipDeprecated = TRUE)` to skip the deprecated check and avoid the problem until I push the fix. (let me know if that works for you!)

**Comment by cjardine-bsc** [comment]
I'm just waiting on a fix to my DB compare tool. I'll work on sorting that out today.

**Comment by steffilazerte** [comment]
No rush, I'll probably get to it mid-next week anyway :grin:

**Comment by cjardine-bsc** [comment]
That poke is for me. I did some syncing last week, but haven't finished. I poked Kwic again about granting access to my IP.

**Comment by steffilazerte** [comment]
Hi @cjardine-bsc, I'm away on vacation for two weeks starting next Wed (June 8th). Any chance we might get this resolved before then? If not, I'll see about adding a quick patch for Josh's problem as a temporary intermediate fix...

**Comment by cjardine-bsc** [comment]
Is it just the additional batches activity column you need (numGPSfix)? In which case, that should be done.

**Comment by steffilazerte** [comment]
@j-sayers The fix has been pushed to the dev branch, did you want to test it out? I'll push to master before I leave on Wednesday, but it's always nice to have it tested by others first!

`remotes::install_github("motusWTS/motus", ref = "dev")`

Should be version v5.0.1 `packageVersion("motus")`

**Comment by j-sayers** [comment]
works!

**Comment by steffilazerte** [comment]
Fix merged to master
